### PR TITLE
#252 add interactions default resolver

### LINKED ISSUE #252: Expose interactions using default resolver
Interactions are resolved via the RxNav API, which returns relatively big payloads and which doesn't support batch queries. This is problematic as it causes query delays and may cause users to misuse the RxNav API and risk the UC server being throttled or blacklisted.

The current solution is that interactions are only exposed via the `entity` query path. For now, this is handled by only exposing interaction as a field via the root `entity` query. 

## Expected Behaviour

Same as now.

## Current Behaviour

Interactions are exposed via a field resolver which contains a conditional which checks the query path for the root query. 

## Possible Solution

Migrate the field resolver to a function returned on the `entity` query (see Apollo docs on [default resolvers]( https://www.apollographql.com/docs/apollo-server/data/resolvers/#default-resolvers)).

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #252.

Branched off #251.

## Description

Migrates data source + resolver logic to data source classes.

### Documentation:

Tick below if one of the following applies:

1. Documentation has been added to cover changes introduced by this PR.
2. This PR requires no changes to documentation.

- [ ] Documentation OK.

### Tests:

Tick below if one of the following applies:

1. Tests have been added to cover changes introduced by this PR.
2. This PR requires no new tests.

- [ ] Tests OK.

### CODE REVIEW

File: unified-codes/apps/data-service/src/app/types/DgraphDataSource.ts [hunk 1]
```diff
@@ -75,9 +81,12 @@ export class DgraphDataSource extends RESTDataSource {
...
     );
 
-    const { counters, query } = data ?? {};
-    const entities: IEntity[] = query ?? [];
-    const [{ total: totalCount }] = counters ?? [];
+    const { counters: counterData, query: entityData } = data ?? {};
+    const [totalCount] = counterData?.total ?? [];
+    // Overwrite interactions to prevent large query delays.
+    const entities: IEntity[] =
+      entityData?.map((entity: IEntity) => ({ ...entity, interactions: [] })) ?? [];
```
Comment (katherine-sussol) [inline unified-codes/apps/data-service/src/app/types/DgraphDataSource.ts:88]: Oh nice 😄

**Review by katherine-sussol — APPROVED** [review]
Sweet 🍬  looks good to me.

Thanks for another 🐈  artifact tidy up 🧹 👍

File: unified-codes/libs/data/src/lib/types/Entity.ts [hunk 2]
```diff
@@ -56,6 +58,16 @@ export class Entity implements IEntity {
     return this._properties;
   }
 
+  getChild(code: string) {
+    const [entity] = this._children?.filter((child: IEntity) => child.code === code);
```
Comment (katherine-sussol) [inline unified-codes/libs/data/src/lib/types/Entity.ts:62]: Oh sorry - minor thing I just noticed from TS on this line (but it does still run fine):
`Type 'Entity[] | undefined' must have a '[Symbol.iterator]()' method that returns an iterator.`

and a similar one for getProperty

Comment (wlthomson) [reply unified-codes/libs/data/src/lib/types/Entity.ts]: Hmmm, c718a7985c9f22442b3f9637d0e0f64caced16f3 should fix that!

**Pull request merged**